File: Solutions/1423/1423.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2021/2/6 8:45
# File: 1423.py
# Desc: 
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        total = sum(cardPoints[:k])
        max_num = total
        for i in range(k):
            total = total - cardPoints[k-i-1] + cardPoints[-i-1]
            if total > max_num:
                max_num = total
        return max_num

File: Solutions/1423/1423.java
/**
 * @Author: washing
 * @DateTime: 2021/2/6 8:45
 * @File: 1423.java
 * @Desc: 
 **/
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int total = 0;
        for(int i=0; i<k; i++){
            total += cardPoints[i];
        }
        int max_num = total;
        int length = cardPoints.length;
        for(int i=0; i<k; i++){
            total = total - cardPoints[k-i-1] + cardPoints[length-i-1];
            if (total > max_num) max_num = total;
        }
        return max_num;
    }
}